src/components/Modal/BaseModal.js

```diff
@@ -76,20 +80,25 @@ function BaseModal({
     );
 
     useEffect(() => {
-        Modal.willAlertModalBecomeVisible(isVisible);
-
-        // To handle closing any modal already visible when this modal is mounted, i.e. PopoverReportActionContextMenu
-        Modal.setCloseModal(isVisible ? onClose : null);
-    }, [isVisible, onClose]);
+        isVisibleRef.current = isVisible;
+        if (isVisible) {
+            Modal.willAlertModalBecomeVisible(true);
+            // To handle closing any modal already visible when this modal is mounted, i.e. PopoverReportActionContextMenu
+            Modal.setCloseModal(onClose);
+        } else if (wasVisible && !isVisible) {
+            Modal.willAlertModalBecomeVisible(false);
```

Comment:
Hi @bernhardoj, we are thinking of moving this line, and the same line from the next `useEffect` into the `hideModal` function. Was there any specific reason not to do it in this PR? Just asking so that we won't break any edge case that I'm not aware of...